pkg/distributor/distributor.go

```diff
@@ -696,12 +694,12 @@ func (d *Distributor) prePushHaDedupeMiddleware(next push.Func) push.Func {
 
 		req, err := pushReq.WriteRequest()
 		if err != nil {
-			return err
+			return distributor_error.NewDistributorPushError(err)
```

Comment:
What is the point of `NewDistributorPushError` wrapping?

Reply:
My idea was to ensure that **all** the errors returned by distributors write path are of type `DistributorError` (that will be renamed in `distributorerror.Error`).

Reply:
The only place on distributor's write path where a cast to a `DistributorError` is not done is a place where the errors from ingesters are returned. This will be done in a follow-up PR when ingester errors will be re-organized analogously to this PR.

Reply:
But this is not distributor error. It's error in parsing the request.

Reply:
There are only 2 errors like that:
- `pushReq.WriteRequest()` and
- `tenant.TenantID()`
Do we want to return them as they are? Or do we want to create another error type (e.g., InternalDistributorPushError`) in `distributorerror` package, to be used for wrapping these 2 errors?

Reply:
As far as I see these errors are treated as InternalServerError, so what's the different between not wrapping them? 

I'd leave the internal errors without wrapping as there's nothing you can do about them and it doesn't add more value. 

As a firstcomer to the code, I expected to see a special behavior associated to this wrapping, but there's none.

Reply:
> I'd leave the internal errors without wrapping as there's nothing you can do about them and it doesn't add more value.

👍

Reply:
I will return the 2 errors as they are, without wrapping them.